### PR TITLE
sway: add config.seat

### PR DESCRIPTION
**Pull request by globin**
### Description

Adds seat configuration for sway

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: tests/modules/services/window-managers/sway/sway-default.conf [hunk 1]
```diff
@@ -71,6 +71,7 @@ bindsym Mod1+w layout tabbed
 
 
 
+
```
Comment (alexarice) [inline tests/modules/services/window-managers/sway/sway-default.conf:74]: I wonder if it's worth changing stripping extra newlines out of the config, as at the moment the generated config doesn't look nice. @rycee does any other module do this?

**Review by alexarice — APPROVED** [review]
Seems reasonable to me

File: tests/modules/services/window-managers/sway/sway-default.conf (hunk 1 above)
Comment (rycee) [reply tests/modules/services/window-managers/sway/sway-default.conf:74]: @alexarice Typically, this is done by concatenating a list of strings. I think it should be pretty easy to change in this case, though. The module uses a lot of code along the lines of
```nix
''
  ${concatStringsSep "\n" (mapAttrsToList inputStr input)}
  ${concatStringsSep "\n" (mapAttrsToList outputStr output)}
  ${concatStringsSep "\n" (mapAttrsToList modeStr modes)}
''
```
which produces these empty lines, if the code instead looks something like
```nix
''
  ${concatStringsSep "\n" (
      mapAttrsToList inputStr input
      ++ mapAttrsToList outputStr output
      ++ mapAttrsToList modeStr modes
  )}
''
```
then it shouldn't produce empty lines.

File: modules/services/window-managers/i3-sway/sway.nix [hunk 2]
```diff
@@ -166,6 +166,15 @@ let
         '';
       };
 
+      seat = mkOption {
+        type = types.attrsOf (types.attrsOf types.str);
+        default = { };
+        example = { "*" = { hide_cursor = "0000"; }; };
+        description = ''
+          An attribute set that defines seats. See man sway_input for options.
```
Comment (rycee) [inline modules/services/window-managers/i3-sway/sway.nix:174]:
```suggestion
          An attribute set that defines seats. See
          <citerefentry>
            <refentrytitle>sway-input</refentrytitle>
            <manvolnum>5</manvolnum>
          </citerefentry>
          for options.
```

File: tests/modules/services/window-managers/sway/sway-default.conf (hunk 1 above)
Comment (sumnerevans) [reply tests/modules/services/window-managers/sway/sway-default.conf:74]: Would it make sense to have this be part of a different PR?

**Comment by sumnerevans** [comment]
@globin are you still planning on making the finishing touches on this? If not, I'd be willing to take over this PR.

**Pull request closed without merging**